projects/plugins/jetpack/extensions/blocks/recurring-payments/test/edit.js

```diff
@@ -0,0 +1,154 @@
+/**
+ * @jest-environment jsdom
+ */
+
+/**
+ * External dependencies
+ */
+import '@testing-library/jest-dom/extend-expect';
+import { render, screen, waitFor } from '@testing-library/react';
+
+/**
+ * Internal dependencies
+ */
+import { MembershipsButtonEdit } from '../edit';
+
+describe( 'MembershipsButtonEdit', () => {
+	const defaultAttributes = {
+		planId: null,
+	};
+
+	const setAttributes = jest.fn();
+	const updateBlockAttributes = jest.fn();
+	const onRemove = jest.fn();
+	const onReplace = jest.fn();
+	const autosaveAndRedirect = jest.fn();
+
+	const defaultProducts = [
+		{
+			id: 1,
+			currency: "USD",
+			price: "10.00",
+			interval: "1 month",
+			title: "ten a month",
+		},
+		{
+			id: 2,
+			currency: "EUR",
+			price: "5.00",
+			interval: "1 year",
+			title: "five a year",
+		}
+	];
+	const defaultProps = {
+		attributes: defaultAttributes,
+		setAttributes,
+		updateBlockAttributes,
+		onRemove,
+		onReplace,
+		autosaveAndRedirect,
+		clientId: 1,
+		postId: 1,
+		noticeList: [],
+		isSelected: true,
+		context: { },
+	};
+
+	const defaultApiResponse = {
+		connect_url: '',
+		connected_account_id: 1,
+		products: defaultProducts,
+		should_upgrade_to_access_memberships: false,
+		site_slug: 'test',
+		upgrade_url: 'https://wordpress.com/checkout/test/jetpack_security_daily_monthly',
+	};
+
+	jest.mock( '../../../shared/get-jetpack-extension-availability', () => () => ( {
+		available: false,
+		unavailableReason: 'missing_plan',
+	} ) );
```

Comment:
This is not working at all; if this were mocked successfully, it would cause the upgrade nudge in the `the upgrade nudge displays` test to not display. I've also set logs in the component and verified that it's getting `unknown` for the `unavailableReason`.

Reply:
Instead of mocking `getJetpackExtensionAvailability`, I think you should be able to add the expected value to the `window` object.

In `get-jetpack-data` there's a constant for `JETPACK_DATA_PATH` which sets it to `window.Jetpack_Editor_Initial_State` — this is where Jetpack renders the initial state server-side for the initial pageload.

So, in our tests, we could try something like:

```
import { JETPACK_DATA_PATH } from '../../../shared/get-jetpack-data';
window[ JETPACK_DATA_PATH ] = {
  available_blocks: {
    'recurring-payments': {
      available: true
    } 
  }
};
```

(If it's a bit clunky using `JETPACK_DATA_PATH`, you can try replacing it with the hard-coded `Jetpack_Editing_Initial_State`)